batcher/batcher.go

```diff
@@ -173,7 +187,13 @@ func (b *basicBatcher) IsDisposed() bool {
 // flush adds the batch currently being built to the queue of completed batches.
 // flush is not threadsafe, so should be synchronized externally.
 func (b *basicBatcher) flush() {
-	b.batchChan <- b.items
+	// Note: This needs to be in a go-routine to avoid locking out gets when
+	// the batch channel is full.
+	cpItems := make([]interface{}, len(b.items))
+	for i, val := range b.items {
+		cpItems[i] = val
+	}
+	go func() { b.batchChan <- cpItems }()
```

Comment:
This wouldn't necessarily preserve the order of items coming in versus going out, as go routines could add the batches in any order once they get backed up, or really before too.